### PR TITLE
feat(Button): Class name & specificity refactor

### LINKED ISSUE #605: [V1] Button refactor
- [ ] split-menu rename to split-button

- [ ] specificity max 0.3.0

- [ ] have single class versus modifiers (i.e. iui-button-high-visibility))

- [ ] Icon button component. Should be square

- [ ] Button group sub components 

- [ ] Borderless active button needs active stripe

### PR DESCRIPTION
**Pull request by FlyersPh9**
## Description

- Specificity reduced from 0,6,0 to 0,3,1.  If we remove / rework `@mixin iui-notification-marker-hover` specificity drops to 0,3,0.
- Replaced `cta.scss` & `high-visibility.scss` with a single mixin that could be used for both button types.
- Use `data-iui-variant`, `data-iui-size`, `data-iui-active` instead of class modifiers.
- `.iui-button` is set to default button & size, no `data-iui-variant` or `data-iui-size` needed.
- `.iui-button-split-menu` renamed to `.iui-button-split`.
- Separated active state from `@mixin iui-button`, is now only present in `@mixin iui-button-default` & `@mixin iui-button-borderless`.
- Better `<a class="iui-button>` support.
- Adding `[aria-disabled='true']` support (mostly for `<a>` support but also helpful for disabled buttons that need focus).
- Removed `.iui-button-dropdown` from header avatar button as it is unused.

Did not include change for 1/1 aspect ratio on borderless buttons with only an icon ([details here](https://github.com/iTwin/iTwinUI/issues/605#issuecomment-1157926363)). Closes #605.

### Button variants & types

- `<button class="iui-button iui-default">` -> `<button class="iui-button">`
- `<button class="iui-button iui-high-visibility">` -> `<button class="iui-button" data-iui-variant="high-visibility">`
- `<button class="iui-button iui-cta">` -> `<button class="iui-button" data-iui-variant="cta">`
- `<button class="iui-button iui-borderless">` -> `<button class="iui-button" data-iui-variant="borderless">`
- `<button class="iui-button iui-idea">` -> `<button class="iui-button" data-iui-variant="idea">`
- `<button class="iui-button iui-dropdown">` -> `<button class="iui-button iui-button-dropdown">`

### Button sizes

- `<button class="iui-button iui-small">` -> `<button class="iui-button" data-iui-size="small">`
- `<button class="iui-button iui-large">` -> `<button class="iui-button" data-iui-size="large">`

### Active button state

- `<button class="iui-button iui-active">` -> `<button class="iui-button" data-iui-active="true">`

### Disabled button states

```diff
<button disabled>
<button disabled="disabled">
+ <button aria-disabled="true">
```

### Button label

- `<span class="iui-button-label">Hello</span>` -> `<span>Hello</span>"`

### Split menu

- `<span class="iui-button-split-menu">` -> `<span class="iui-button-split">`

### CODE REVIEW

File: packages/itwinui-css/src/button/split-button.scss [hunk 1]
```diff
@@ -2,46 +2,46 @@
...
-    &:first-child {
-      > .iui-borderless:hover {
-        background-color: transparent;
-      }
-    }
+    // &:first-child {
+    //   > [data-iui-button-variant='borderless']:hover {
+    //     background-color: transparent;
+    //   }
+    // }
```
Comment (FlyersPh9) [inline packages/itwinui-css/src/button/split-button.scss:17]: Unsure what lines 13-17 are used for.

File: packages/itwinui-css/backstop/tests/button.html [hunk 2]
```diff
@@ -1248,14 +1577,22 @@ <h3>Borderless</h3>
 
     <h3>Split menu</h3>
     <section id="demo-split-menu">
```
Comment (adamhock) [inline packages/itwinui-css/backstop/tests/button.html:1579]: Nitpicking here, but should we change this id to `demo-split` if we're changing `iui-button-split-menu` to `iui-button-split`?

File: packages/itwinui-css/backstop/tests/button.html [hunk 3]
```diff
@@ -1470,21 +1875,30 @@ <h3>Input + button combo group</h3>
     <section id="demo-combo-group">
       <div class="iui-button-group">
         <div>
-          <button class="iui-button iui-default"><span class="iui-button-label">Button 1</span></button>
+          <button
```
Comment (adamhock) [inline packages/itwinui-css/backstop/tests/button.html:1878]: HTML for these four buttons could be cleaned up as well

**Review by adamhock — COMMENTED** [review]
Changes look good to me so far!

File: packages/itwinui-css/src/button/split-button.scss (hunk 1 above)
Comment (adamhock) [reply packages/itwinui-css/src/button/split-button.scss:17]: This code was for the background color for split buttons in the header.

Before:
SplitMenuChangesBefore


After: 
SplitMenuChangesAfter

File: packages/itwinui-css/backstop/tests/button.html (hunk 3 above)
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/button.html:1878]: Fixed.

File: packages/itwinui-css/backstop/tests/tile.html [hunk 4]
```diff
@@ -647,7 +688,13 @@ <h2>Location</h2>
         </div>
 
         <div class="iui-tile-buttons">
-          <button class="iui-button iui-default">Manage</button><button class="iui-button iui-default">Projects</button>
+          <button
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/tile.html:697]: Fixed.

File: packages/itwinui-css/src/button/classes.scss [hunk 5]
```diff
@@ -4,49 +4,44 @@
...
 
-    &.iui-borderless {
-      @include iui-button-size(large, borderless);
-    }
-  }
+.iui-button-split {
+  @include iui-button-split;
 }
 
 .iui-button-group {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/classes.scss:47]: Updated `classes.scss` to not use the `&-` method.

File: packages/itwinui-css/src/button/idea.scss [hunk 6]
```diff
@@ -1,8 +1,9 @@
 // Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 // See LICENSE.md in the project root for license terms and full copyright notice.
+
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/idea.scss]: Empty line removed.

File: packages/itwinui-css/backstop/tests/button.html (hunk 2 above)
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/button.html:1579]: For now I've only updated the test names & test ID.

File: packages/itwinui-css/src/button/split-button.scss [hunk 7]
```diff
@@ -50,9 +50,4 @@
   &.iui-disabled {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/split-button.scss:50]: This class is applied as a modifier on `.iui-button-split` which is applied on a `<span>`, not a `<button>`.  So `:disabled, [disabled]` wouldn't work here, but we could possibly add `[aria-disabled='true']` for consistency.

File: packages/itwinui-css/src/style/global.scss [hunk 8]
```diff
@@ -66,8 +66,8 @@ html {
   @include iui-scrollbar;
 }
 
-[class*=iui-]:where(:not(.iui-body)),
-[class*=iui-]:where(:not(.iui-body))::before,
-[class*=iui-]:where(:not(.iui-body))::after {
+[class*='iui-']:where(:not(.iui-body)),
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/style/global.scss:69]: I don't recall making this change, maybe the linter did this?

File: packages/itwinui-css/backstop/tests/menu.html [hunk 9]
```diff
@@ -637,7 +637,8 @@ <h3>Header</h3>
           <nav aria-label="breadcrumbs">
             <!-- Project Button -->
             <button
-              class="iui-button iui-borderless iui-large iui-header-button iui-header-dropdown-button"
+              class="iui-button iui-header-button iui-header-dropdown-button"
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/menu.html:640]: I just tested with `.iui-small` & `.iui-large`, am seeing no change.  The button size is definitely not needed in the breadcrumbs after #640.

File: packages/itwinui-css/src/notification-marker/notification-marker.scss [hunk 10]
```diff
@@ -16,7 +16,8 @@ $iui-notification-marker-border-width: $iui-xxs;
     right: $iui-sm * -0.5;
     border-radius: 100%;
     background-color: var(--iui-color-foreground-#{$status});
-    border: $iui-notification-marker-border-width solid var(--iui-color-background-1);
+    border: $iui-notification-marker-border-width solid var(--iui-color-background-1); // Fallback border color incase user forgets to set var(--_iui-notification-marker-border-color) on parent.
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/notification-marker/notification-marker.scss:19]: `var(--_iui-notification-marker-border-color)` needs to be set on any parent component that could have the notification marker within it.  So far that means the borderless & default button.  But if a user wanted to use the notification marker within a custom component, they _should_ set this variable.  In the chance that they don't know to do so or forget, I offer a fallback.

[the remark this answers is not in the document]
Comment (veekeys) [reply packages/itwinui-css/src/notification-marker/notification-marker.scss:19]: So still.. why we cannot assign default value to the variable instead of default border value?

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/notification-marker/notification-marker.scss]: Oh derp, you are right, updated.

File: packages/itwinui-css/src/style/global.scss (hunk 8 above)
[the remark this answers is not in the document]
Comment (veekeys) [reply packages/itwinui-css/src/style/global.scss:69]: amm.. that is super weird

**Review by veekeys — APPROVED** [review]
I was not originally assigned to this PR, but i havent noticed anything bad.
Would be good to remember cases, when it was hard to reuse button for some new components and see, if this refactor helps in any way.

File: packages/itwinui-css/backstop/tests/button.html [hunk 11]
```diff
@@ -241,35 +290,53 @@ <h3>Call to action</h3>
         id="demo-cta-small"
         class="demo-container"
       >
-        <button class="iui-button iui-cta iui-small">
+        <button
+          class="iui-button"
+          data-iui-button-variant="cta"
+          data-iui-button-size="small"
+        >
           <span class="iui-button-label">Small</span>
```
Comment (veekeys) [inline packages/itwinui-css/backstop/tests/button.html:298]: By the way, you have renamed this in header, should we align???

Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/button.html:298]: In #640 I am using `class="iui-header-breadcrumb-button"` seen [here](https://github.com/iTwin/iTwinUI/blob/a8eb0e12ff8f1703b2867fdb158af74283508c83/packages/itwinui-css/backstop/tests/header.html#L122).  Are you suggesting something like `<button class="iui-button" data-iui-button-variant="header-breadcrumb">`?

Comment (veekeys) [reply packages/itwinui-css/backstop/tests/button.html:298]: In header we have 
``` 
text
  label
  sublabel
```

In Button we have just `label`. So we might want to align the naming, at least

File: packages/itwinui-css/src/button/button.scss [hunk 12]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
   }
 
-  @include iui-focus;
+  &[data-iui-button-size='large'] {
+    @include iui-button-size(large);
+  }
+  // #endregion Sizing
+
+  &[aria-disabled='true'] {
+    @include iui-focus;
```
Comment (mayank99) [inline packages/itwinui-css/src/button/button.scss:50]: Shouldn't `iui-focus` be included for all states rather than _only_ for disabled?

File: packages/itwinui-css/src/button/button.scss [hunk 13]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
   &[disabled],
-  &:disabled {
-    @include iui-button-disabled;
+  &:disabled,
+  &[aria-disabled='true'] {
+    cursor: not-allowed;
+
+    &,
+    &:hover,
+    &:active {
```
Comment (mayank99) [inline packages/itwinui-css/src/button/button.scss:60]: Is there any reason to add all these extra selectors?

Ideally, disabled always comes at the end and overrides any previous states. If specificity is an issue, maybe we should use `:where` in all the places that we change `:hover` states.

File: packages/itwinui-css/src/button/split-button.scss [hunk 14]
```diff
@@ -2,46 +2,46 @@
...
+      > [data-iui-button-variant='cta'] {
         border-right-color: rgba(var(--iui-color-foreground-accessory-rgb), var(--iui-opacity-4));
       }
     }
 
     &:last-child {
-      > .iui-high-visibility,
-      > .iui-cta {
+      > [data-iui-button-variant='high-visibility'],
+      > [data-iui-button-variant='cta'] {
```
Comment (mayank99) [inline packages/itwinui-css/src/button/split-button.scss:44]: All the overrides in this file look like a good opportunity to use CSS variables.

File: packages/itwinui-css/src/button/index.scss [hunk 15]
```diff
@@ -2,11 +2,8 @@
 // See LICENSE.md in the project root for license terms and full copyright notice.
 @import './button';
 @import './default';
-@import './cta';
-@import './high-visibility';
+@import './borderless';
+@import './status-color';
```
Comment (mayank99) [inline packages/itwinui-css/src/button/index.scss:6]: So I'm getting a little confused by these file names, particularly `button`, `default` and `status-color`.

Is there a reason we need to keep `button` and `default` separate? I feel like the default styling should be part of the base button.

Also, in another comment I suggested renaming `status-color` to `variants`.

File: packages/itwinui-css/src/button/classes.scss [hunk 16]
```diff
@@ -4,67 +4,54 @@
...
 
-.iui-button-split-menu {
-  @include iui-button-split-menu;
+.iui-button-group-vertical {
+  @include iui-button-group(vertical);
 }
 
-.iui-button-icon {
-  @include iui-button-icon;
+.iui-button-group-overflow-x {
```
Comment (mayank99) [inline packages/itwinui-css/src/button/classes.scss:55]: Why changed this?

File: packages/itwinui-css/src/button/index.scss (hunk 15 above)
Comment (mayank99) [reply packages/itwinui-css/src/button/index.scss:6]: Also where did `button-icon` disappear?

File: packages/itwinui-css/src/button/classes.scss [hunk 17]
```diff
@@ -4,67 +4,54 @@
...
 
-  &.iui-borderless {
-    @include iui-button-borderless;
-  }
+[data-iui-button-variant='borderless'] {
+  @include iui-button-borderless;
+}
 
-  @include iui-button-size(medium);
+[data-iui-button-variant='high-visibility'] {
```
Comment (mayank99) [inline packages/itwinui-css/src/button/classes.scss:26]: Can you explain your thought process on `data-iui-button-variant` vs `data-iui-variant`? Same for `data-iui-button-sizing`.

File: packages/itwinui-css/backstop/tests/button.html (hunk 2 above)
Comment (adamhock) [reply packages/itwinui-css/backstop/tests/button.html]: I would say either keep it at `Split menu` or just simply change it to `Split` to be consistent with other sections (Default, Borderless, etc.)

File: packages/itwinui-css/src/button/classes.scss (hunk 17 above)
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/classes.scss:26]: My thinking was I could make the data attribute more specific and keep the specificity lower:
```scss
.iui-button { … }
[data-iui-button-variant] { … }
```
Versus this which has a higher specificity:
```scss
.iui-button {
  &[data-iui-variant] { … }
}
```

File: packages/itwinui-css/src/notification-marker/notification-marker.scss [hunk 18]
```diff
@@ -5,6 +5,8 @@
 $iui-notification-marker-border-width: $iui-xxs;
 
 @mixin iui-notification-marker($status: primary) {
+  --_iui-notification-marker-border-color: var(--iui-color-background-1);
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/notification-marker/notification-marker.scss]: Reverted all notification marker changes.

File: packages/itwinui-css/backstop/tests/button.html (hunk 11 above)
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/button.html:298]: Replaced `.iui-button-label` with `.iui-button-text`.

File: packages/itwinui-css/src/button/button.scss [hunk 19]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
+      --_iui-button-text-color: var(--iui-text-color-placeholder);
+      --_iui-button-background-color: var(--iui-color-background-disabled);
+      --_iui-button-border-color: var(--iui-color-background-disabled);
+    }
+  }
+}
+
+@mixin iui-button-label {
+  white-space: nowrap;
+  color: inherit;
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: I've removed the mixin but left the class names even though they don't have any styling associated.

File: packages/itwinui-css/src/button/button.scss [hunk 20]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
+
+@mixin iui-button-icon {
+  width: $iui-icons-default;
+  height: $iui-icons-default;
+  transition: fill $iui-speed-fast ease-out;
+  fill: currentColor;
+
+  @at-root [data-iui-button-variant='borderless'] &:only-child {
+    margin-left: $iui-component-offset;
+    margin-right: $iui-component-offset;
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: I like this, change made!

File: packages/itwinui-css/src/button/status-color.scss [hunk 21]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import './button';
+
+@mixin iui-button-status-color($button-color) {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/status-color.scss]: Renamed the mixin to `iui-button-variant` & the file name to match.

I did not make the change yet to accept `high-visibility` or `cta` as the args yet.

File: packages/itwinui-css/src/color-picker/color-picker.scss [hunk 22]
```diff
@@ -76,7 +76,7 @@ $iui-hover-box-shadow: 0 0 1px $iui-xxs + 1 rgba(var(--iui-color-foreground-body
   min-width: $iui-l;
   gap: $iui-s;
 
-  button.iui-button.iui-borderless {
+  button.iui-button[data-iui-button-variant='borderless'] {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/color-picker/color-picker.scss]: Done.  Did not add a variable for width because that's not typically set.  Could add it, but didn't so far.

File: packages/itwinui-css/src/button/button.scss [hunk 23]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
+  background-color: var(--_iui-button-background-color);
   color: var(--_iui-button-text-color);
-  @media (prefers-reduced-motion: no-preference) {
-    transition: color $iui-speed-fast ease-out, background-color $iui-speed-fast ease-out,
-      border-color $iui-speed-fast ease-out;
+  transition: background-color $iui-speed-fast ease-out, border-color $iui-speed-fast ease-out;
+  -webkit-tap-highlight-color: transparent;
+
+  @at-root a#{&} {
+    text-decoration: none;
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss:38]: Ok, done.

File: packages/itwinui-css/src/button/button.scss (hunk 19 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-css/src/button/button.scss]: Where is the class name useful? Could we replace those usages with `> *`?

File: packages/itwinui-css/src/button/button.scss (hunk 20 above)
[the remark this answers is not in the document]
Comment (mayank99) [reply packages/itwinui-css/src/button/button.scss]: I meant toggling the value of the variable like this (you've done the opposite):

```scss
--_iui-button-padding-inline: #{$iui-component-offset};
```

**Review by adamhock — APPROVED** [review]
I approve once all the other conversations are resolved

File: packages/itwinui-css/backstop/tests/button.html (hunk 2 above)
Comment (FlyersPh9) [reply packages/itwinui-css/backstop/tests/button.html]: If I make this change I will make it very last so as to not introduce test image changes.

File: packages/itwinui-css/src/button/button.scss [hunk 24]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
   }
 
-  &:hover {
-    --_iui-button-text-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-1));
+  // #region Sizing
+  @include iui-button-size;
 
-    text-decoration: none;
+  &[data-iui-button-size='small'] {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss:41]: Which do you prefer?

1. Sizes are applied explicitly on the variant types, no overrides. Does cause other components (breadcrumb buttons, paginator buttons) to break but can be fixed by including the mixin.
Screen Shot 2022-07-08 at 12 22 09 PM
2. Sizes are set on `.iui-button` and override on `[data-iui-button-variant='borderless']`.
Screen Shot 2022-07-08 at 12 23 21 PM

File: packages/itwinui-css/src/button/button.scss (hunk 12 above)
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss:51]: `iui-focus` is set on the variant types due to differences between the variants:
- Default button = 2px outline
Screen Shot 2022-07-08 at 12 28 30 PM
- Borderless button = 1px outline
Screen Shot 2022-07-08 at 12 28 48 PM
- CTA & High visibility button = 1px outline, -1px inset
Screen Shot 2022-07-08 at 12 29 14 PM Screen Shot 2022-07-08 at 12 29 26 PM



Then for all `aria-disabled="true"` buttons, no matter the variant type, we reset the `iui-focus` to 1px outline.
Screen Shot 2022-07-08 at 12 30 13 PM

File: packages/itwinui-css/src/button/classes.scss (hunk 16 above)
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/classes.scss:51]: We've been trying to avoid `&-` in a bunch of our CSS 1.0 PRs so that it's easier to search & find classes.

If I search for `.iui-button-group-overflow-x` it won't find this:
```scss
.iui-button-group {
  &-overflow-x { … }
}
```

File: packages/itwinui-css/src/button/button.scss (hunk 19 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: Within this button PR it looks like `.iui-button-text` is being styled within `header.scss` and `side-navigation.scss`. I don't think `> *` would work in those instances as is currently because there is also an icon nested within the button.

However, #640 uses [`.iui-header-breadcrumb-button-text`](https://github.com/iTwin/iTwinUI/blob/b5143acda0da28517bff191ec7273e0dd1548fb1/packages/itwinui-css/backstop/tests/header.html#L133) instead.  We could do the same to `side-navigation`.

[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: @veekeys, any opinion on this?  [You suggested using `.iui-button-text`](https://github.com/iTwin/iTwinUI/pull/684#discussion_r914013136) to align with other components.

File: packages/itwinui-css/src/button/default.scss [hunk 25]
```diff
@@ -2,50 +2,39 @@
...
-    position: relative;
-    color: var(--iui-color-foreground-primary);
-
-    // Used for combining overlay background color with base background
-    &::before {
-      content: '';
-      position: absolute;
-      inset: 0;
-      background-color: rgba(var(--iui-color-foreground-primary-rgb), var(--iui-opacity-6));
+  &.iui-active {
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/default.scss]: This is done.  It required some changes in tile, paginator, info panel to use the mixin, CSS vars, etc.

File: packages/itwinui-css/src/button/button.scss [hunk 26]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
-    &::after {
-      background-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-5));
+@mixin iui-button-dropdown {
+  padding-right: $iui-s;
+
+  &[data-iui-button-size='small'] {
+    padding-right: $iui-xs;
+
+    &[data-iui-button-variant='borderless'] {
+      padding-right: $iui-xxs;
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss:106]: I've updated `variant.scss` to accept the button variant name as `@arg` instead of color.  Personally I think the older way was cleaner code wise but the `@arg`s make more sense this new way.

File: packages/itwinui-css/src/button/status-color.scss (hunk 21 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/status-color.scss]: I've updated `variant.scss` to accept the button variant name as `@arg` instead of color.  Personally I think the older way was cleaner code wise but the `@arg`s make more sense this new way.

File: packages/itwinui-css/src/button/default.scss [hunk 27]
```diff
@@ -2,50 +2,39 @@
...
-@import './button-icon';
 @import '../notification-marker/index';
-@import './disabled';
 
 @mixin iui-button-default {
-  background-color: var(--iui-color-background-1);
-  border-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-4));
+  --_iui-button-text-color: var(--iui-text-color);
+  --_iui-button-background-color: var(--iui-color-background-1);
+  --_iui-button-border-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-4));
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/default.scss:10]: But the text color uses opacity-2 and borders use opacity-4.  If `border-opacity` was a thing it would be possible but as it is right now I don't think it will work.

File: packages/itwinui-css/src/button/button.scss (hunk 13 above)
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss:59]: Can you explain how I can use `:where` to fix this? I'm not understanding.

File: packages/itwinui-css/src/color-picker/color-picker.scss (hunk 22 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/color-picker/color-picker.scss]: Removed width, not needed.

File: packages/itwinui-css/src/button/button.scss [hunk 28]
```diff
@@ -22,49 +19,99 @@ $iui-button-padding-large: $iui-xs * 6;
...
+    --_iui-button-text-color: var(--iui-color-foreground-primary);
+  }
+
   // Shows active stripe only if --_iui-button-active-stripe-inset is set (from outside)
   &::after {
     content: '';
     position: absolute;
     inset: var(--_iui-button-active-stripe-inset);
-    background-color: var(--iui-color-foreground-primary);
+    background-color: currentColor;
```
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: Stripe color is hardcoded, no CSS var.

File: packages/itwinui-css/src/button/button.scss (hunk 20 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: Resolved over call.

File: packages/itwinui-css/src/button/button.scss (hunk 19 above)
[the remark this answers is not in the document]
Comment (FlyersPh9) [reply packages/itwinui-css/src/button/button.scss]: Will remove `.iui-button-text` classes.

**Review by mayank99 — APPROVED** [review]
Approving this as API (class names) looks good. @FlyersPh9 Please make sure to document every single change here in the PR description as well as in a comment in #639 

From code perspective, there are three questionable parts:
- Split button / button group looks messy (tackle in #706?)
- Sizing modifiers are all over the place (fixable in the future without changing the attributes)
- Some CSS variables are hard to follow and are creating unnecessary depth (fixable in future without breaking API). e.g. carousel dots use internal button variables to set height and padding. Ideally, these internal variables should not be used like token (to reference values) but rather for toggling the characteristics of the button by assigning some other value _to_ it.
